syntax = "proto2";

package ml;

import "ml/model.proto";
import "ml/optimizer/optimizer.proto";

// Linear regression model parameters.
// Next id: 11
message LinearRegressionParameters {
  // Maximum number of iterations.
  optional int64 max_iter = 1;

  // Minimum number of iterations.
  optional int64 min_iter = 2;

  // Stopping criterions.
  optional double eps = 3 [default = 0.0001];

  // Learning rate (fixed step length; ignored if the backtracking line search
  // is enabled).
  optional double gamma = 4 [default = 0.0001];

  // Regression coefficients.
  repeated double parameters = 8;

  // Optimization settings.
  optional optimizer.OptimizerParameters optimizer_parameters = 9;

  extend ModelParameters {
    optional LinearRegressionParameters options = 101;
  }
};
